src/compiler/checker/lines_8000_12000.rs

```diff
@@ -144,7 +153,24 @@ impl TypeChecker {
         type_
     }
 
+    pub(super) fn get_type_of_instantiated_symbol(&self, symbol: &Symbol) -> Rc<Type> {
+        let links = self.get_symbol_links(symbol);
+        let mut links = links.borrow_mut();
+        if links.type_.is_none() {
+            let type_ = self.instantiate_type(
```

Comment:
This is actually what will return a `number` type by looking up the `T` type parameter in the `TypeMapper` that maps `T` -> `number`